### ISSUE TITLE
Vault Timeout is not honored on Android - it locks immediately and forces me to authenticate everytime I need to autofill a password.

### ISSUE THREAD

**Issue by jmannrider** [bitwarden/android#1557]
### Steps To Reproduce

1. Go to Bitwarden->Settings->Vault Timeout
2. Select any timeout option other than immediately. I'll select 'Never'
3. Go to any app that allows you to auto-fill a password
4. Unlock vault and fill password in.
5. Go to any other app and autofill password.
6. Notice that the vault is already locked and must authenticate to unlock it again.

### Expected Result

I would expect to not have to authenticate again if 'Never' is selected as the timeout option.

### Actual Result

Vault locks immediately despite the option you've selected.

### Additional Context

I have troubleshooted this by making sure that the phone's battery settings is not killing the app. Bitwarden is set to never be put to sleep or optimized at all. I have already ran out of options.
I have also contacted support but they weren't able to offer me a solution.

### Operating System

Android

### Operating System Version

9

### Web Browser

N/A- Only app.

### Browser Version

N/A 

### Build Version

2.12.0(3742)

**Comment by gschintgen** [comment]
I can confirm (on Android) that I'm constantly asked to unlock my vault even though I tried setting the timeout to 4 hours.
Version: 2.16.2 (4334)

**Comment by gschintgen** [comment]
Update: it *does* work for me after all if I disable battery savings:
Settings/Apps/Bitwarden/Battery/Unrestricted
This is on a Samsung Galaxy S10e (Android 12)

Sorry for the noise...

**Comment by ps2goat** [comment]
This just started happening to me today. I've had this set up on my phone since Jan 2022 and have used it every day since. I'm on a Samsung A32(?). Version 2023.9.1

Every time I navigate away or need to authority, I have to sign in again.

**Comment by ImSkully** [comment]
Also experiencing this issue since some time in August, there is a [forum topic](https://community.bitwarden.com/t/vault-locks-automatically-when-timeout-is-disabled-in-android-app/58506/11) *(and #2182)* on this issue with many other users reporting the same.

Rolling back to version [2023.8.0](https://www.apkmirror.com/apk/bitwarden-inc/bitwarden-password-manager/bitwarden-password-manager-2023-8-0-release/bitwarden-password-manager-2023-8-0-2-android-apk-download/) seems to fix this so a change released after this has re-introduced this problem again, at least for me anyway.

**Comment by vinnyperella** [comment]
This is still an issue I just updated to 2023.10.0 which states in the notes "Bug fix for timeout settings not being respected" but I'm still seeing the issue. If I set my vault timeout to 5 minutes, swipe Bitwarden out of my recent apps, then open Bitwarden again it prompts me for my PIN regardless of the timeout setting. If I keep Bitwarden in my recent apps after 5 minutes it prompts me for my PIN which is the correct behavior. Not sure if this is intended I can't remember which version this started occurring on but I know it wasn't always the case.

This also screws up the "enter master password when app is restarted" feature I used to have enabled. If you enable that and swipe Bitwarden out of your recent apps it will prompt you for your master password. It used to only prompt for your master password if the app was updated or the device was restarted. This is super annoying as this is a very good feature for security that I simply cannot use anymore.

**Comment by vinnyperella** [comment]
Any update on this issue?

**Comment by meiersa78** [comment]
Just started using Bitwarden and this issue is driving me mad. Apparently, it's been years and nothing has been done... When will there be a fix for Android? Reddit is full of posts about this very problem. I can set any time, custom time, never, but BW just locks me out. I disabled any battery optimization for the app. I even locked the app so it never gets closed by Android, but still, BW just locks the vault all the time, no matter if set to 5 minutes, 4 hours or never. This is so frustrating! What's the point of that setting if it doesn't work at all?! Ah, and I tried biometric unlock and the pin method. I also cleared data and cache, I uninstalled and reinstalled the app countless times. It doesn't work!
